example/eblob_backend.cpp

```diff
@@ -747,95 +748,206 @@ struct iterated_key_info {
-			json = data_pointer::allocate(info.jhdr.size);
-			const int err = dnet_read_ll(info.fd, json.data<char>(), json.size(), info.json_offset);
+		if (info->jhdr.size) {
+			json = data_pointer::allocate(info->jhdr.size);
+			const int err = dnet_read_ll(info->fd, json.data<char>(), json.size(), info->json_offset);
 			if (err) {
 				dnet_backend_log(c->blog, DNET_LOG_ERROR,
 				                 "EBLOB: server_send: %s: failed to read json: %s",
-				                 dnet_dump_id_str(info.key.id), dnet_print_error(err));
+				                 dnet_dump_id_str(info->key.id), dnet_print_error(err));
 				return 0;
 			}
 		}
 
 		data_pointer data;
-		if (info.data_size) {
-			data = data_pointer::allocate(info.data_size);
-			const int err = dnet_read_ll(info.fd, data.data<char>(), data.size(), info.data_offset);
+		if (info->data_size) {
+			data = data_pointer::allocate(info->data_size);
+			const int err = dnet_read_ll(info->fd, data.data<char>(), data.size(), info->data_offset);
 			if (err) {
 				dnet_backend_log(c->blog, DNET_LOG_ERROR,
 				                 "EBLOB: server_send: %s: failed to read data: %s",
-				                 dnet_dump_id_str(info.key.id), dnet_print_error(err));
+				                 dnet_dump_id_str(info->key.id), dnet_print_error(err));
 				return 0;
 			}
 		}
 
 		auto session = std::make_shared<newapi::session>(st->n);
 		session->set_exceptions_policy(session::no_exceptions);
 		session->set_trace_id(cmd->trace_id);
-		session->set_trace_id(!!(cmd->flags & DNET_FLAGS_TRACE_BIT));
+		session->set_trace_bit(!!(cmd->flags & DNET_FLAGS_TRACE_BIT));
 		session->set_groups(request.groups);
-		session->set_user_flags(info.ehdr.flags);
+		session->set_user_flags(info->ehdr.flags);
 		session->set_ioflags(DNET_IO_FLAGS_CAS_TIMESTAMP);
-		// if (dnet_time_cmp(&info.ehdr.timestamp, &info.jhdr.timestamp) < 0) {
-		// 	session->set_timestamp(info.jhdr.timestamp);
+		// if (dnet_time_cmp(&info->ehdr.timestamp, &info->jhdr.timestamp) < 0) {
+		// 	session->set_timestamp(info->jhdr.timestamp);
 		// } else {
-			session->set_timestamp(info.ehdr.timestamp);
+			session->set_timestamp(info->ehdr.timestamp);
 		// }
 		if (session->get_timeout() < 60) {
 			session->set_timeout(60);
 		}
 
-		auto async = session->write(info.key,
-		                            json, info.jhdr.capacity,
-		                            data, info.data_size);
+		monitor.add_bytes(info->jhdr.size + info->data_size);
+
+		auto async = session->write(info->key,
+		                            json, info->jhdr.capacity,
+		                            data, info->data_size);
 
-		async.connect([&] (const newapi::sync_write_result &/*results*/, const error_info &error) {
+		async.connect([=, &request, &monitor] (const newapi::sync_write_result &/*results*/, const error_info &error) {
 			if (st->__need_exit) {
 				dnet_backend_log(c->blog, DNET_LOG_ERROR,
 				                 "EBLOB: Interrupting server_send: peer has been disconnected");
```

Comment:
There should be return statement after the log. Could you add it and move `monitor.remove_bytes(info->jhdr.size + info->data_size);` before `if (st->__need_exit) {`?